src/psyclone/psyir/transformations/loop_fuse_trans.py

```diff
@@ -103,18 +125,184 @@ def validate(self, node1, node2, options=None):
             if not (SymbolicMaths.equal(node1.start_expr, node2.start_expr) and
                     SymbolicMaths.equal(node1.stop_expr, node2.stop_expr) and
                     SymbolicMaths.equal(node1.step_expr, node2.step_expr)):
+                # TODO #257: This transformation assumes that all domain loop
+                # bodies have only POINTWISE accesses to fields and does not
+                # perform any dependency analysis.
+                # This is wrong and it will generate incorrect code for any
+                # kernel with STENCIL access.
                 raise TransformationError(LazyString(
                     lambda node1=node1, node2=node2:
                         f"Error in {self.name} transformation. Loops do not "
                         f"have the same iteration space:\n"
-                        f"{node1.view()}\n"
-                        f"{node2.view()}"))
+                        f"{node1.debug_string()}\n"
+                        f"{node2.debug_string()}"))
+
+            vars1 = VariablesAccessInfo(node1)
+            vars2 = VariablesAccessInfo(node2)
+
+            # Check if the loops have the same loop variable
+            loop_var1 = node1.variable
+            loop_var2 = node2.variable
+            if loop_var1 != loop_var2:
+                # If they don't have the same variable, find out if the one
+                # loop accesses the other loops variable symbol.
+                # If so, then for now we disallow this merge (though we could
+                # in theory allow using the unused one unless both use each
+                # others)
+                for signature in vars1:
```

Comment:
Sorry for adding to an ongoing review ... but I just noticed that the loops can be removed by using:
```
if Signature(loop_var2.name) in vars1:
   raise ...
```
(and similarly for the next loop)